homeassistant/components/synology_dsm/sensor.py

```diff
@@ -132,6 +132,10 @@ def should_poll(self) -> bool:
         """No polling needed."""
         return False
 
+    async def async_update(self):
+        """Only used by the generic entity update service."""
+        await self._api.update()
```

Comment:
Tested and working, but not sure to be a good idea.

At start:

```shell
2020-05-03 23:16:33 INFO (MainThread) [homeassistant.components.sensor] Setting up sensor.synology_dsm
2020-05-03 23:16:38 INFO (MainThread) [homeassistant.setup] Setting up device_tracker
2020-05-03 23:16:42 INFO (MainThread) [homeassistant.setup] Setup of domain device_tracker took 3.9 seconds.
2020-05-03 23:16:42 INFO (MainThread) [homeassistant.components.device_tracker] Setting up device_tracker.netgear
2020-05-03 23:16:43 WARNING (SyncWorker_23) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_12) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_6) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_37) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_26) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_19) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_10) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_20) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_15) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_14) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_18) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_39) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_31) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_24) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_13) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_2) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_21) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_30) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_3) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_5) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_1) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_32) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_7) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_17) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_11) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (SyncWorker_0) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:43 WARNING (MainThread) [homeassistant.components.sensor] Setup of sensor platform synology_dsm is taking over 10 seconds.
2020-05-03 23:16:44 WARNING (SyncWorker_16) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:44 WARNING (SyncWorker_0) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:44 WARNING (SyncWorker_11) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:44 WARNING (SyncWorker_28) [urllib3.connectionpool] Connection pool is full, discarding connection: nas.XXXX.com
2020-05-03 23:16:44 INFO (MainThread) [homeassistant.bootstrap] Home Assistant initialized in 20.87s
2020-05-03 23:16:44 INFO (MainThread) [homeassistant.core] Starting Home Assistant
```

My `Synology CPU Load (1 min)` went to 1022%.

Reply:
Remove `True` to `async_add_entities` line 64, seems to solve the issue :wink:

Reply:
Good call out. I made your suggested change, tested, and verified I'm not seeing high CPU load on setup. Looked into the method call, and it appears that the second parameter on `async_add_entities` forces an update on sensor add, which was doing it for each and every sensor.